### PR TITLE
627 send claims events to cx dw

### PR DESCRIPTION
**Pull request by helene-mccarron**
## Description of change

Given the security concerns of transmitting PII to SharePoint (see https://github.com/department-of-veterans-affairs/va-virtual-agent/issues/615), we have pivoted to only sending non-PII data to our longer term Omnichannel data warehouse: CxDW from vets-api. We send this data when users request for claims info in the chatbot. VBA will be responsible to access this data from CxDW once we place it over there.

## Original issue(s)
department-of-veterans-affairs/va-virtual-agent#627

## Things to know about this PR

- Added new secrets to `settings.yml`

Unit tests added and VCR cassettes added with secrets removed.

### CODE REVIEW

File: app/controllers/v0/virtual_agent/report_to_cxdw.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+module V0
+  module VirtualAgent
+    class ReportToCxdw
+      def report_to_cxdw(icn, conversation_id)
+        dataverse_uri = Settings.cxdw_dataverse_uri
+        token = get_new_token dataverse_uri
+        send_to_cxdw dataverse_uri, icn, conversation_id, token
+      end
+
+      def get_new_token(dataverse_uri)
```
Comment (holdenhinkle) [inline app/controllers/v0/virtual_agent/report_to_cxdw.rb:17]: Perhaps make `#get_new_token` a `private` method?

File: app/controllers/v0/virtual_agent/report_to_cxdw.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        request = Net::HTTP::Post.new(url)
+        request['content-type'] = 'application/x-www-form-urlencoded'
+        request.body = "#{body_grant_type}&#{body_client_id}&#{body_client_secret}&#{body_scope}/.default"
+
+        response = http.request(request)
+        JSON.parse(response.read_body)['access_token']
+      end
+
+      def send_to_cxdw(dataverse_uri, icn, conversation_id, token)
```
Comment (holdenhinkle) [inline app/controllers/v0/virtual_agent/report_to_cxdw.rb:41]: Perhaps make `#send_to_cxdw` a `private` method?

**Review by holdenhinkle — DISMISSED** [review]
You might want to consider making a couple `V0::VirtualAgent::ReportToCxdw` methods private.

There are also some Rubocop issues to handle.

Other than that, it LGTM!

File: app/controllers/v0/virtual_agent/report_to_cxdw.rb (hunk 2 above)
Comment (khathuynh-tw) [reply app/controllers/v0/virtual_agent/report_to_cxdw.rb:43]: Done!

**Review by holdenhinkle — APPROVED** [review]

**Review by holdenhinkle — APPROVED** [review]

**Pull request merged**